### PR TITLE
fix(datepicker): Check if getter.assign is function fixes #3155

### LINKED ISSUE #3155: Date picker TypeError: setAttribute is not a function setAttribute(scope.$parent, value);
I got error then datepicker is closed after date is selected
Version: 0.12.0 - 2014-11-16
```javascript
TypeError: setAttribute is not a function
setAttribute(scope.$parent, value);
```

### PR DESCRIPTION
**Pull request by rvanbaalen**
Closes #3155

### CODE REVIEW

File: src/datepicker/datepicker.js [hunk 1]
```diff
@@ -490,7 +490,7 @@ function ($compile, $parse, $document, $position, dateFilter, dateParser, datepi
           if ( key === 'datepickerMode' ) {
             var setAttribute = getAttribute.assign;
             scope.$watch('watchData.' + key, function(value, oldvalue) {
-              if ( value !== oldvalue ) {
+              if ( typeof setAttribute === 'function' && value !== oldvalue ) {
```
Comment (wesleycho) [inline src/datepicker/datepicker.js]: Use `angular.isFunction`

**Comment by JohannesRudolph** [comment]
@rvanbaalen would you be able to update your pull request? Should be trivial to merge then.

**Comment by JohannesRudolph** [comment]
I can confirm this fix works using angular.isFunction

**Comment by rvanbaalen** [comment]
Working on it.

**Comment by rvanbaalen** [comment]
Had to create a new PR since I previously trashed my bootstrap fork

**Comment by rvanbaalen** [comment]
Landed in ed10899d6b4dc4ab3c46c6b0288ecd8d98cc852a

**Pull request closed without merging**